### PR TITLE
 Comments feature

### PR DESCRIPTION
**Pull request by janko101**
### Changes proposed
- Adds a feature to update Post
- Adds routing to update Post
- Fix API entry to update post, added ID field
- Adds a feature to Comment on the post
- Add Comment class, Comment form, and Comment List
- Adds a feature to delete a Comment
- Adds a feature to view a specific post and associated comments
- Card updated with link to comments and delete button
- PostDetails page added
- Relationship between User and Comment / Post added
- Adds display of user Name by comments and posts made, along with generic avatar
- Update and refactor of CSS, files split into Layout, Comment, Post

### CODE REVIEW

File: frontend/src/App.jsx [hunk 1]
```diff
@@ -25,17 +26,24 @@ export default function App() {
...
+
           <Route path="/chat">
             <ChatPage />
           </Route>
 
           <Route path="/">
             <HomePage />
           </Route>
+
+
```
Comment (janko101) [inline frontend/src/App.jsx:45]: too many spaces...We need to format

File: frontend/src/api/CommentsAPI.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import Api from "./Api";
+
+class CommentsApi {
+  getAllComments(postId) {
+    return Api.get("/posts/" + postId + "/comments");
+  }
+
+// Currently we don't have this mapping request. 
+//   getCommentById(id) {
+//     return Api.get("/comments/" + id);
```
Comment (janko101) [inline frontend/src/api/CommentsAPI.js:10]: Delete comment code

File: build.gradle [hunk 3]
```diff
@@ -27,6 +27,8 @@ dependencies {
 	developmentOnly 'org.springframework.boot:spring-boot-devtools'
 	compile group: 'com.auth0', name: 'java-jwt', version: '3.8.3'
 	runtimeOnly 'org.postgresql:postgresql'
+	testImplementation 'org.apache.httpcomponents:httpclient'
+
```
Comment (clecardona) [inline build.gradle:31]: we don't need that but ok

File: frontend/src/api/PostsApi.js [hunk 4]
```diff
@@ -1,25 +1,25 @@
...
-    }
+  createPost(post) {
+    return Api.post("/posts", post);
+  }
 
-    updatePost(post) {
-        return Api.put('/posts', post);
-    }
+  updatePost(id, post) {
+    return Api.put(`/posts/${id}`, post);
```
Comment (janko101) [inline frontend/src/api/PostsApi.js:17]: This will haunt me for days :))

**Review by clecardona — APPROVED** [review]

File: build.gradle (hunk 3 above)
Comment (deciocandido) [reply build.gradle:31]: Does not interact the testing

File: frontend/src/App.jsx (hunk 1 above)
Comment (deciocandido) [reply frontend/src/App.jsx:45]: Refacturing better in the future

File: frontend/src/api/PostsApi.js (hunk 4 above)
Comment (deciocandido) [reply frontend/src/api/PostsApi.js:17]: passing the ID as ref to PUT

**Comment by deciocandido** [comment]
Some of the comments were to refacturing and clean coding

**Pull request merged**